### PR TITLE
Hooks and split

### PR DESCRIPTION
**Pull request by 0ssigeno**
Splitted kernel32.
Added more hooks.
I swear I will provide a test once a sample works entirely.

### CODE REVIEW

**Comment by xwings** [comment]
Hi,

for all the examples, we leave the "import qiling" after the sys.path.append("..") is due to we can run the example without install.

I guess we should keep it there.

**Comment by 0ssigeno** [comment]
Reverted as asked!

**Comment by xwings** [comment]
i think Quynh just updated the pe.py. can you fix it ?

**Comment by 0ssigeno** [comment]
Should be good on the conflict now

**Comment by xwings** [comment]
i guess we need some time to review it. so far so good ! but still allow us to take some time

**Comment by 0ssigeno** [comment]
Let me know if there are any problems

File: examples/adcache_x86_windows_debug.py [hunk 1]
```diff
@@ -4,15 +4,15 @@
 # Built on top of Unicorn emulator (www.unicorn-engine.org)
 
 import sys
-from zipfile import ZipFile
-
 sys.path.append("..")
+from zipfile import ZipFile
```
Comment (xwings) [inline examples/adcache_x86_windows_debug.py]: this should be on top of sys.path.append ?

File: examples/hello_mips32el_linux_debug.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 # Built on top of Unicorn emulator (www.unicorn-engine.org)
 
 import sys, string, random
-sys.path.append("..")
 from qiling import *
+sys.path.append("..")
```
Comment (xwings) [inline examples/hello_mips32el_linux_debug.py]: this should be ontop of from qiling import and not below

File: examples/hello_arm_linux_debug.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 # Cross Platform and Multi Architecture Advanced Binary Emulation Framework
 # Built on top of Unicorn emulator (www.unicorn-engine.org) 
 import sys
-sys.path.append("..")
 from qiling import *
+sys.path.append("..")
```
Comment (xwings) [inline examples/hello_arm_linux_debug.py]: thiis one too

File: examples/hello_arm_linux_custom_syscall.py [hunk 4]
```diff
@@ -4,8 +4,8 @@
 # Built on top of Unicorn emulator (www.unicorn-engine.org)
 
 import sys
-sys.path.append("..")
 from qiling import *
+sys.path.append("..")
```
Comment (xwings) [inline examples/hello_arm_linux_custom_syscall.py]: same for append should be on to of qiling import

**Comment by xwings** [comment]
Thankyou sir !

**Pull request merged**